### PR TITLE
Api trending news

### LINKED ISSUE #141: Integrate API in trending news

### PR DESCRIPTION
**Pull request by ravverma**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #141 

Test URLs:
- Before: https://main--icicidirect--aemsites.hlx.live/research/equity
- After: https://api-trending-news--icicidirect--aemsites.hlx.live/research/equity

### CODE REVIEW

File: blocks/trending-news/trending-news.js [hunk 1]
```diff
@@ -51,9 +64,9 @@ function createNewsCards(news) {
   const iconSpan = document.createElement('span');
   iconSpan.className = 'icon icon-icon-time';
   const abbr = document.createElement('abbr');
-  abbr.textContent = `${news.date} `;
+  abbr.textContent = `${item.PublishedOn} `;
   const abbrSource = document.createElement('abbr');
-  abbrSource.textContent = news.source;
+  abbrSource.textContent = item.Author;
```
Comment (anagarwa) [inline blocks/trending-news/trending-news.js:69]: @ravverma do we get this information from API ?

Comment (ravverma) [reply blocks/trending-news/trending-news.js:69]: yes, Author is in the API response.

**Review by anagarwa — APPROVED** [review]

File: blocks/trending-news/trending-news.js [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
-  Viewport, createPictureElement, observe, fetchData,
+  getResearchAPIUrl, postFormData,
+  Viewport, createPictureElement, observe, parseResponse,
 } from '../../scripts/blocks-utils.js';
 import { decorateIcons, fetchPlaceholders, readBlockConfig } from '../../scripts/aem.js';
 
 const placeholders = await fetchPlaceholders();
+const ICICI_DIRECT_NEWS_HOST = 'https://www.icicidirect.com/research/equity/trending-news/';
+const ICICI_NEWS_THUMBNAIL_ICICI_HOST = 'https://www.icicidirect.com/images/';
+const ICICI_NEWS_THUMBNAIL_FINOUX_HOST = 'http://icicidirect.finoux.com/images/';
```
Comment (anagarwa) [inline blocks/trending-news/trending-news.js:10]: @ravverma  may be we should extract these into block utils

Comment (anagarwa) [reply blocks/trending-news/trending-news.js:10]: @ravverma for http://icicidirect.finoux.com , we can fetch the images from edge worker as done it here
https://github.com/aemsites/icicidirect/pull/148/files#diff-042cbe0e2ef3c10e1d0e56d1416e393d9eca4b7fcdbe69ea03322e7b14303d97R106.

and for https://www.icicidirect.com IMO we should get a confirmation that this is really the correct way. We should be assuming that all this is from UAT environment of their and hence that should be the host url.
We can ask again.

Comment (sdmcraft) [reply blocks/trending-news/trending-news.js:10]: Why do we need these? Isn't the api response returning absolute URLs? If not, we should ask.

Comment (ravverma) [reply blocks/trending-news/trending-news.js:10]: No, API doesn't retun absolute , not even relative it is just return resource part of url only.

Comment (ravverma) [reply blocks/trending-news/trending-news.js:10]: yes , I will add in tracker

**Review by sdmcraft — APPROVED** [review]
Building urls etc. seems fragile to me and ideally we should get full URLs from API. Let track it via open questions and merge this for now.

**Pull request merged**